generators/newview/index.js

```diff
@@ -1,96 +1,155 @@
 
-        var sOrigin = this.templatePath(sViewFileName);
-        var sTarget = this.destinationPath(sViewFileName.replace(/\$ViewEnding/, sViewType.toLowerCase()).replace(/\$ViewName/, sViewName));
-        this.fs.copyTpl(sOrigin, sTarget, this.options.oneTimeConfig);
+		const sViewType = this.options.oneTimeConfig.viewtype;
+		const sViewName = this.options.oneTimeConfig.viewname;
 
-        if (this.options.oneTimeConfig.createcontroller || this.options.isSubgeneratorCall) {
-            sOrigin = this.templatePath(sControllerFileName);
-            sTarget = this.destinationPath(sControllerFileName.replace(/\$ViewEnding/, sViewType.toLowerCase()).replace(/\$ViewName/, sViewName));
-            this.fs.copyTpl(sOrigin, sTarget, this.options.oneTimeConfig);
-        }
-    }
+		var sOrigin = this.templatePath(sViewFileName);
+		var sTarget = this.destinationPath(sViewFileName.replace(/\$ViewEnding/, sViewType.toLowerCase()).replace(/\$ViewName/, sViewName));
+		this.fs.copyTpl(sOrigin, sTarget, this.options.oneTimeConfig);
 
-    end() {
-        if (this.options.isSubgeneratorCall) {
-            return;
+		if (this.options.oneTimeConfig.createcontroller || this.options.isSubgeneratorCall) {
+			sOrigin = this.templatePath(sControllerFileName);
+			sTarget = this.destinationPath(sControllerFileName.replace(/\$ViewEnding/, sViewType.toLowerCase()).replace(/\$ViewName/, sViewName));
+			this.fs.copyTpl(sOrigin, sTarget, this.options.oneTimeConfig);
+		}
+
+	}
+
+	end() {
+		if (this.options.isSubgeneratorCall) {
+			return;
         }
-        this.log('Created a new view!');
-    }
-};
\ No newline at end of file
+        
+        const localOptions = this.options;
+        const sViewType = this.options.oneTimeConfig.viewtype;
+		const sViewName = this.options.oneTimeConfig.viewname;
+		async function f() {
+
+			let promise = new Promise((resolve, reject) => {
+				if (localOptions.oneTimeConfig.addToRoute) {
+					const filePath = process.cwd() + '/webapp/manifest.json';
+					fs.readFile(filePath, function (err, data) {
+						let json = JSON.parse(data)
+						let ui5Config = json['sap.ui5'],
+							routing = ui5Config.routing,
+							routes = routing.routes;
+
+						const routeName = 'Route' + sViewName,
+							targetName = 'Target' + sViewName;
+
+						let route = {
+							name: sViewName,
+							pattern: routeName,
+							target: [targetName]
+						}
+						routes.push(route);
+
+						routing.targets[targetName] = {
+							viewType: sViewType,
+							viewName: sViewName
+						};
+
+						ui5Config.routing = routing;
+						json['sap.ui5'] = ui5Config;
+
+						fs.writeFile(filePath, JSON.stringify(json, null, 4), function (err) {
+							if (err) throw err;
+							resolve('Updated manifest file with routing, remember to update the pattern');
+						});
+					})
+				} else {
+					reject();
+				}
+			});
+
+			let result = await promise; // wait until the promise resolves (*)
+
+			console.log(result); // "done!"
```

Comment:
The [documentation suggests](https://yeoman.io/authoring/user-interactions.html) that we shouldn't use the console object in yeoman